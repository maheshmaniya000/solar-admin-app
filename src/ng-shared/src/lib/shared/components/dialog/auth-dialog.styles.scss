@import 'theme';

// This is special styling for dialogs
// It is put here so that it is included only once in the bundled version of app

:host ::ng-deep .auth-dialog-content {
  font-size: $sg-14;
  line-height: 20px;

  h3 {
    font-size: $sg-18;
    line-height: 22px;
    margin-bottom: 16px;
    font-weight: $font-weight-bold;
  }

  p {
    margin-bottom: 16px;

    a {
      text-decoration: underline;
      cursor: pointer;

      &:hover {
        text-decoration: none;
      }

      &.accent-link {
        color: map-get($accent, default);
        text-decoration: none;

        &:hover {
          text-decoration: underline;
        }
      }
    }
  }

  .margin-bottom {
    margin-top: 16px;
  }

  .padding-left {
    padding-left: 29px;
  }

  .inline-paragraph {
    padding-top: 0.84375em;
    margin-bottom: 0;
    line-height: 42px;
  }

  .recaptcha-container {
    padding: 0;

    & > sg-captcha {
      display: block;
      margin: 20px 0;
    }
  }

  ::ng-deep {
    .mat-form-field {
      display: block;
      position: relative;

      input, mat-select {
        @include sg-input;

        &:disabled {
          background-color: map-get($primary, 200);
          border-color: map-get($primary, 200);
        }
      }
    }

    .mat-form-field-label {
      top: 10px;
      position: relative;
      color: map-get($primary, 800);
      font-weight: $font-weight-medium;
    }

    .mat-form-field-invalid .mat-form-field-label {
      color: map-get($primary, 800);
    }

    .mat-warn .mat-input-element, .mat-form-field-invalid .mat-input-element,
    .mat-warn mat-select, .mat-form-field-invalid mat-select {
      border: 1px solid map-get($accent, 500);
    }

    .mat-form-field-wrapper {
      padding-bottom: 0;
    }

    .mat-form-field-appearance-legacy .mat-form-field-underline {
      height: 0;
      background: none;
    }

    .mat-form-field-appearance-legacy.mat-form-field-invalid:not(.mat-focused) .mat-form-field-ripple {
      height: 0;
    }

    .mat-form-field-subscript-wrapper {
      margin-top: 0;
    }

    .mat-form-field-ripple {
      background: none;
      height: 0;
    }

    .mat-form-field-subscript-wrapper {
      position: relative;
    }

    .mat-input-element {
      width: 100%;
      box-sizing: border-box;
    }

    .mat-horizontal-stepper-header-container {
      display: none;
    }

    .mat-horizontal-content-container {
      padding: 0;
    }

    .mat-form-field-suffix {
      position: absolute;
      right: 8px;
      top: 24px;

      .mat-button-wrapper {
        font-size: 20px;
        color: map-get($primary, 600);
      }

      .mat-icon-button {
        outline: none;
      }
    }
  }

  .loading {
    margin: 0 auto 16px auto;

    > mat-spinner {
      display: inline-block;
    }
  }

  .error {
    margin-bottom: 15px;
  }

  .mat-error {
    color: map-get($accent, default);
  }

  .success {
    .success-list {
      padding: 0;
      list-style: none;

      li {
        color: #008E09;
        line-height: 17px;
        margin-bottom: 12px;

        mat-icon {
          color: white;
          background-color: #008E09;
          border-radius: 50%;
          margin-right: 12px;
          float: left;
          font-size: 17px;
          width: 17px;
          height: 17px;
          line-height: 17px;
        }

        &.error {
          color: map-get($accent, default);

          mat-icon {
            background-color: map-get($accent, default);
          }
        }
      }
    }
  }

  .invitation-summary {
    border-top: 1px solid map-get($primary, 300);
    padding-top: 16px;

    .field {
      color: map-get($primary, 600);
    }

    .info {
      padding-top: 32px;

      mat-icon {
        color: map-get($accent, default);
        width: 20px;
        height: 20px;
        line-height: 20px;
        float: left;
        margin-right: 12px;
      }
    }
  }
}

// Dialog actions
:host ::ng-deep .auth-dialog-actions {
  button {
    line-height: 18px;
    width: 100%;
    margin-bottom: 12px;

    &:last-child {
      margin-bottom: 0;
    }

    @include media-breakpoint-up(sm) {
      float: right;
      width: auto;
      margin-bottom: 0;

      &:not(:first-child) {
        margin-right: 12px;
      }
    }
  }

  .mat-button-wrapper {
    text-transform: uppercase;
  }

  .primary { @include sg-button(lg, red); }
  .secondary { @include sg-button(lg); }

  .back-button {
    overflow: hidden;
    background: none;
    border: 0;
    margin: 0;
    padding: 0;
    min-width: auto;
    height: 40px;
    line-height: 40px;
    color: map-get($primary, 700);

    @include media-breakpoint-up(sm) {
      float: left;
    }

    mat-icon {
      float: left;
      display: block;
      margin: 8px 8px 8px 0;
    }

    span {
      float: left;
    }
  }
}

// Dialog footer
:host ::ng-deep .auth-dialog-footer {
  padding: 12px 32px 0 32px;
  border-top: 1px solid map-get($primary, 250);
  font-size: $sg-14;
  line-height: 20px;

  p {
    margin-bottom: 12px;
  }

  a {
    color: map-get($accent, default);

    &:hover {
      color: map-get($accent, default);
      text-decoration: underline;
    }
  }
}
