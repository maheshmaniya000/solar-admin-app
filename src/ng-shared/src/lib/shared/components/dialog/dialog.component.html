<div class="dialog-container bootstrap" [ngClass]="{'large': large, 'with-min-width': !!minWidth}"
    [style.minWidth.px]="minWidth">
  <div class="dialog-wrapper d-flex flex-column h-100">

    <div mat-dialog-title class="d-flex">
      <div class="title d-flex flex-wrap">
        <span class="close-dialog">
          <ng-content select=".sg-dialog-close-button"></ng-content>
        </span>
        <span class="heading">
          <ng-content select=".sg-dialog-heading"></ng-content>
        </span>
        <ng-content select=".sg-dialog-heading-content"></ng-content>
      </div>
    </div>
    <mat-dialog-content #contentWrapper [class.scroll-header]="isScroolHeader" class="flex-fill d-flex flex-wrap table-wrapper">
      <div #content class="content">
        <ng-content select=".sg-dialog-content"></ng-content>
      </div>
    </mat-dialog-content>

    <mat-dialog-actions class="d-flex">
      <div class="actions">
          <hr *ngIf="showSeparator">
        <ng-content select=".sg-dialog-actions"></ng-content>
      </div>

      <div class="footer">
        <ng-content select=".sg-dialog-footer"></ng-content>
      </div>
    </mat-dialog-actions>
  </div>
</div>
