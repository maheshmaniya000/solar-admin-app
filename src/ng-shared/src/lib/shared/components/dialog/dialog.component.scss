@import 'theme';

// has an impact globally.
::ng-deep {
  .cdk-overlay-pane {
    max-width: 100vw !important;
  }

  .mat-dialog-content {
    max-height: calc(90vh - 200px) !important;
  }
}

@include media-breakpoint-down(sm) {
  ::ng-deep { // Making the modal fullscreen in mobiles
    .cdk-overlay-pane {
      max-height: 100vh !important;
    }

    .dialog-container:not(.with-min-width) .mat-dialog-content {
      max-height: unset !important;
    }
  }

  .dialog-wrapper {
    width: 100% !important;
    min-width: 320px !important;
  }
}

.dialog-container {
  .dialog-wrapper {
    padding-bottom: $sg-14;
  }

  width: 100%;
  height: 100vh;

  &.with-min-width {
    width: auto !important;
    height: auto;
  }

  & > div {
    overflow: hidden;
  }

  @include media-breakpoint-up(md) {
    width: 560px;
    height: auto;

    &.large {
      width: auto;
    }
  }
}

:host ::ng-deep {
 color: map-get($primary, 900);

  .title {
    font-size: $sg-18;
    line-height: 24px;
    padding: 24px 32px 0 32px;
    overflow: hidden;

    .heading {
      font-weight: $font-weight-bold;
    }

    .close-dialog:not(:empty) {
      margin-right: 16px;

      button {
        width: 24px;
        height: 24px;
        line-height: 24px;
        float: left;
        outline: none;

        mat-icon {
          color: map-get($primary, 800);
          float: left;
          line-height: 24px;
        }
      }

      .mat-button-focus-overlay {
        background: none;
        transition: none;
      }
    }
  }

  .content {
    padding: 0 32px;
    width: 100%;
  }

  mat-dialog-actions {
    min-height: unset;
    justify-content: flex-end;
  }

  .actions {
    width: 100%;
    min-height: 36px;
    padding: 0 32px 24px 32px;
      hr {
        margin: 0 0 10px 0;
        border-top: 1px solid rgba(0, 0, 0, 0.12);
    }

    button {
      line-height: 36px;
      margin-left: 12px;
    }
  }

  .with-min-width {
    .actions button {
      float: right;
      width: auto;
      margin-bottom: 0;

      &:not(:first-child) {
        margin-right: 12px;
      }
    }
  }

  .actions:empty {
    display: none;
  }

  mat-dialog-content, .mat-dialog-actions:last-child, .mat-dialog-title {
    margin: 0;
    padding: 0;
  }


  .mat-dialog-title {
    margin-top: 14px;
  }

  mat-dialog-content {
    margin: 12px 0 0;
  }

  .footer {
    width: 100%;

    span {
      font-size: $sg-14;
      line-height: 18px;
      color: map_get($accent, 500);
    }
  }

  .footer:empty {
    display: none;
  }

  .mat-form-field-invalid {
    .mat-form-field-label {
      color: map-get($accent, 500);
    }
    .mat-form-field-ripple {
      background-color: map-get($accent, 500);
    }
  }
}

@media (min-width: 760px) {
  .mat-dialog-content {
    margin: 34px 0 0 0;
  }

  mat-dialog-content, .mat-dialog-actions:last-child, .mat-dialog-title {
    margin: 5px 0 0 0;
    padding: 0;
  }


  .dialog-container {
    .dialog-wrapper {
      padding-bottom: 0;
    }
  }

  :host ::ng-deep {
    .sg-dialog-actions {
      margin: 40px 0 0 0;

    }
    .mat-dialog-title {
      margin-top: 5px;
    }
  }
}

::ng-deep {
  .sg-dialog-actions {
    display: flex;
    justify-content: flex-end;
  }
}

.scroll-header ::ng-deep .mat-tab-header{
  background-color: white;
  position: sticky;
  top: 0;
  z-index: 2;
}
