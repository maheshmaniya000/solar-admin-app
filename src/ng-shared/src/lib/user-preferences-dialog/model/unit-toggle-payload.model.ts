export type UnitTogglePayload = [{ settingsKey: string; toggleKey: string }];
