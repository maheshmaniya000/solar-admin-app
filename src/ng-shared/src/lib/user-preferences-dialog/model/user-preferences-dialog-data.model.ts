import { StandardDialogData } from 'components/dialog/standard-dialog/model/standard-dialog-data.model';

import { UserPreferencesDialogInput } from './user-preferences-dialog-input.model';
import { UserPreferencesDialogPayload } from './user-preferences-dialog-payload.model';

export type UserPreferencesDialogData =
  StandardDialogData<[undefined, UserPreferencesDialogPayload]> & UserPreferencesDialogInput;
