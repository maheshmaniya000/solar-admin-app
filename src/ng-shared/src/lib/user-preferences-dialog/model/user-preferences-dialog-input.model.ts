import { UnitToggle, UnitToggleSettings } from '@solargis/units';

import { UserPreferences } from 'components/models/user-preferences.model';


export interface UserPreferencesDialogInput {
  userPreferences: UserPreferences;
  unitToggles: UnitToggle[];
  unitToggleSettings: UnitToggleSettings;
}
