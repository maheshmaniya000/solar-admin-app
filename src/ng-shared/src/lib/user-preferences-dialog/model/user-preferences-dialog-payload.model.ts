import { UserPreferences } from 'components/models/user-preferences.model';

import { UnitTogglePayload } from './unit-toggle-payload.model';

export interface UserPreferencesDialogPayload {
  userPreferences?: UserPreferences;
  unitTogglePayload?: UnitTogglePayload;
}
