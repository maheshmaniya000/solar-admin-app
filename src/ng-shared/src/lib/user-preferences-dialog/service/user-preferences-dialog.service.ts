import { Injectable } from '@angular/core';
import { MatDialogRef } from '@angular/material/dialog';

import { DialogConstants } from 'components/dialog/dialog.constants';
import { DialogService } from 'components/dialog/service/dialog.service';
import { StandardDialogComponent } from 'components/dialog/standard-dialog/standard-dialog.component';

import { UserPreferencesDialogInput } from '../model/user-preferences-dialog-input.model';
import { UserPreferencesDialogPayload } from '../model/user-preferences-dialog-payload.model';
import { UserPreferencesDialogContentComponent } from '../user-preferences-dialog-content.component';
import { UserPreferencesDialogModule } from '../user-preferences-dialog.module';

@Injectable({
  providedIn: UserPreferencesDialogModule
})
export class UserPreferencesDialogService {
  constructor(private readonly dialogService: DialogService) {}

  open(
    data: UserPreferencesDialogInput
  ): MatDialogRef<StandardDialogComponent, UserPreferencesDialogPayload> {
    return this.dialogService.openStandardDialog({
      data: {
        titleTranslationKey: 'Time and unit settings',
        dividedContent: true,
        content: UserPreferencesDialogContentComponent,
        actions: DialogConstants.createStandardDialogDefaultActions(),
        ...data
      }
    });
  }
}
