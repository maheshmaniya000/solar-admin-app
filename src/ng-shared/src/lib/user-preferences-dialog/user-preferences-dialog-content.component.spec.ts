import { MAT_DIALOG_DATA } from '@angular/material/dialog';
import {
  createComponentFactory,
  createSpyObject,
  Spectator,
  SpyObject
} from '@ngneat/spectator/jest';
import { MockComponent } from 'ng-mocks';
import { Subject } from 'rxjs';

import { flattenDataLayers, resolveDataset } from '@solargis/dataset-core';
import { UnitToggleService } from '@solargis/ng-unit-value';
import { UnitToggleSettings } from '@solargis/units';

import { DialogConstants } from 'components/dialog/dialog.constants';
import { Timezone } from 'components/models/timezone.model';
import { UserPreferences } from 'components/models/user-preferences.model';
import { TimezoneInputComponent } from 'components/timezone-input/timezone-input.component';

import { UnitTogglePlainService } from '../plain-services/unit-toggle-plain.service';
import { UserPreferencesDialogData } from './model/user-preferences-dialog-data.model';
import { UserPreferencesDialogPayload } from './model/user-preferences-dialog-payload.model';
import { UserPreferencesDialogContentComponent } from './user-preferences-dialog-content.component';

describe('UserPreferencesDialogContentComponent', () => {
  const dataset = resolveDataset(
    flattenDataLayers({
      ELE: { name: 'Elevation', unit: 'm' },
      GHI: {
        name: 'Global horizontal irradiation',
        resolution: {
          annual: { access: 'free' },
          monthly: { access: 'prospect:lta:paid' }
        },
        unit: ['kWh/m2', '{{resolution}}']
      },
      TEMP: { name: 'Air temperature', unit: '°C' }
    }),
    { prefix: 'test.dataLayer' }
  );

  const createDialogData = (): UserPreferencesDialogData => ({
    title: 'Dialog title',
    content: UserPreferencesDialogContentComponent,
    actions: DialogConstants.createStandardDialogDefaultActions(),
    userPreferences: { timezone: Timezone.site, utcOffset: 2 },
    unitToggles: UserPreferencesDialogContentComponent.dataLayerMapToUnitToggles(
      dataset.annual
    ),
    unitToggleSettings: { temperature: '°F' }
  });

  describe('unit', () => {
    let unitToggleServiceSpyObject: SpyObject<UnitToggleService>;
    let payload: UserPreferencesDialogPayload;
    let component: UserPreferencesDialogContentComponent;

    const mockSelectUnitToggle$ = new Subject<UnitToggleSettings>();

    beforeEach(() => {
      unitToggleServiceSpyObject = createSpyObject(UnitTogglePlainService);
      unitToggleServiceSpyObject.selectUnitToggle.andReturn(
        mockSelectUnitToggle$
      );

      component = new UserPreferencesDialogContentComponent(
        createDialogData(),
        unitToggleServiceSpyObject
      );
      component.ngOnInit();
      payload = component.data.actions[1].payload;
    });

    describe('dataLayerMapToUnitToggles', () => {
      [undefined, null].forEach(value => {
        it(`should throw if ${value} is provided`, () => {
          expect(() =>
            UserPreferencesDialogContentComponent.dataLayerMapToUnitToggles(value)
          ).toThrow();
        });
      });

      it(`should map dataset's data layers to array of unique unit toggles it contains`, () => {
        expect(
          UserPreferencesDialogContentComponent.dataLayerMapToUnitToggles(
            dataset.annual
          )
        ).toEqual(
          ['length', 'irradiation', 'ltaDaily', 'temperature'].map(
            settingsKey => expect.objectContaining({ settingsKey })
          )
        );
      });
    });

    describe('ngOnInit', () => {
      it('should set initial unit toggle settings from the injected data by dispatching unit toggle payload to UnitToggleService', () => {
        expect(
          unitToggleServiceSpyObject.dispatchUnitToggle
        ).toHaveBeenCalledTimes(1);
        expect(
          unitToggleServiceSpyObject.dispatchUnitToggle
        ).toHaveBeenCalledWith([
          { settingsKey: 'temperature', toggleKey: '°F' }
        ]);
      });

      it('should subscribe to form value changes and save emitted values to the payload', () => {
        const userPreferences: UserPreferences = {
          timezone: Timezone.utc,
          utcOffset: 5
        };
        component.form.patchValue(userPreferences);

        expect(payload.userPreferences).toEqual(userPreferences);
      });

      it('should subscribe to UnitToggleService.selectUnitToggle observable and save emitted unit toggle settings to the payload', () => {
        mockSelectUnitToggle$.next({ length: 'ft' });

        expect(
          unitToggleServiceSpyObject.selectUnitToggle
        ).toHaveBeenCalledTimes(1);
        expect(payload.unitTogglePayload).toEqual([
          { settingsKey: 'length', toggleKey: 'ft' }
        ]);
      });
    });
  });

  describe('component', () => {
    let spectator: Spectator<UserPreferencesDialogContentComponent>;

    const createComponent = createComponentFactory({
      component: UserPreferencesDialogContentComponent,
      declarations: [MockComponent(TimezoneInputComponent)],
      providers: [{ provide: MAT_DIALOG_DATA, useValue: createDialogData() }],
      shallow: true
    });

    beforeEach(() => (spectator = createComponent()));

    it('should match snapshot and have correctly bound methods and attributes', () => {
      const timezoneComponent = spectator.query(TimezoneInputComponent);
      const { timezone, utcOffset } = spectator.component.data.userPreferences;

      expect(timezoneComponent.timezone).toEqual(timezone);
      expect(timezoneComponent.utcOffset).toEqual(utcOffset);
      expect(spectator.fixture).toMatchSnapshot();
    });
  });
});
