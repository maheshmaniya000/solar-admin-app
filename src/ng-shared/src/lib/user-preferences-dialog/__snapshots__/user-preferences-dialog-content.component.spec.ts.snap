// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPreferencesDialogContentComponent component should match snapshot and have correctly bound methods and attributes 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
  data={[Function Object]}
  destroyed$={[Function Subject]}
  form={[Function FormGroup]}
  latlngSample={[Function Object]}
  latlngToggle={[Function Object]}
  latlngUnit={[Function Object]}
  unitToggleService={[Function UnitTogglePlainService]}
  unitToggleSettingsKeyToLabel={[Function Object]}
>
  <div
    class="sg-grid-8-15"
  >
    <form>
      <sg-timezone-input
        class="row-6-9"
      />
    </form>
    <div
      class="row-6-9 row-next"
    >
      <span>
        Latitude and longitude
      </span>
      <sg-unit-toggle />
      <sg-unit-value />
    </div>
    <div
      class="row-6-9 row-next"
    >
      <span>
        Elevation and length
      </span>
      <sg-unit-toggle />
    </div>
    <div
      class="row-6-9 row-next"
    >
      <span>
        Solar radiation
      </span>
      <sg-unit-toggle />
    </div>
    <div
      class="row-6-9 row-next"
    >
      <span>
        Long-term averages
      </span>
      <sg-unit-toggle />
    </div>
    <div
      class="row-6-9 row-next"
    >
      <span>
        Temperature
      </span>
      <sg-unit-toggle />
    </div>
  </div>
</ng-component>
`;
